Noetic test dev

Axelito95 opened the pull request:
This is a fix to import leg_detector in noetic version. opencv -> opencv2

DLu commented:
Thanks for working on this. 

Right now the CI isn't working because [bfl](https://wiki.ros.org/noetic/Migration#Use_Orocos_KDL_and_BFL_rosdep_keys) is still in the package.xml .

Axelito95 commented:
> 
> 
> Thanks for working on this.
> 
> Right now the CI isn't working because [bfl](https://wiki.ros.org/noetic/Migration#Use_Orocos_KDL_and_BFL_rosdep_keys) is still in the package.xml .

i don't know what is CI, i saw that in your package you already added bfl migration key [bfl](https://wiki.ros.org/noetic/Migration#Use_Orocos_KDL_and_BFL_rosdep_keys) . i had bfl library problem but i fixed this on my pc installed the library manually : sudo apt install liborocos-bfl-dev

up : i saw that you right in package.xml is bfl depedency o.o, i will fix this soon

DLu commented:
My apologies, CI stands for continuous integration which is the system which ensures the code will compile for everyone.

Axelito95 commented:
> My apologies, CI stands for continuous integration which is the system which ensures the code will compile for everyone.

Np ! just my bad that i don't know CI acronyme.Added new bfl key dependency

i see that the build doesn't pass because : 
Checking for module 'orocos-bfl'
--   No package 'orocos-bfl' found

On my pc i fixed it with : sudo apt-get install -y liborocos-bfl-dev

DLu commented on leg_detector/src/calc_leg_features.cpp:
```diff
@@ -34,8 +34,10 @@
 
-#include <opencv/cxcore.h>
-#include <opencv/cv.h>
+//#include <opencv/cxcore.h>
+//#include <opencv/cv.h>
```
```suggestion
```

DLu commented on leg_detector/src/leg_detector.cpp:
```diff
@@ -37,9 +37,11 @@
-#include <opencv/cxcore.h>
-#include <opencv/cv.h>
-#include <opencv/ml.h>
+//#include <opencv/cxcore.hpp>
+//#include <opencv/cv.h>
+//#include <opencv/ml.h>
```
```suggestion
```

Axelito95 replied:
yes sure, you can! I let it just to show to others what libraries can be replaced but I saw that they can see the changes from commits

DLu commented on leg_detector/src/leg_detector.cpp:
```diff
@@ -96,12 +99,13 @@ class SavedFeature
 
-  Stamped<Point> position_;
+  tf::Stamped<tf::Point> position_;
+  //Stamped<Point> position_;
```
```suggestion
```

DLu commented on leg_detector/src/train_leg_detector.cpp:
```diff
@@ -35,9 +35,11 @@
-#include <opencv/cxcore.h>
-#include <opencv/cv.h>
-#include <opencv/ml.h>
+//#include <opencv/cxcore.hpp>
+//#include <opencv/cv.h>
+//#include <opencv/ml.h>
```
```suggestion
```

DLu reviewed (commented):
@Axelito95 Do you mind if I remove these commented lines?

DLu commented:
Thank you very much @Axelito95 for figuring this out!

Axelito95 commented:
> 
> 
> Thank you very much @Axelito95 for figuring this out!

You are welcome ! And thank you too for your working ! Happy new year ;) !
